src/accelerate/utils/modeling.py

```diff
@@ -127,13 +127,11 @@ def set_module_tensor_to_device(
 
         if is_buffer:
             module._buffers[tensor_name] = new_value
-        elif module._parameters[tensor_name].device.type != "cuda":
+        elif device != module._parameters[tensor_name].device:
```

Comment:
```suggestion
        elif device != module._parameters[tensor_name].device.index:
```
The `device` param is an integer so to properly compare the devices we need to access the index of the `device` attribute of the parameter ;)

Reply:
Ah should be `torch.device(device)` on the left side to cover all cases!

Reply:
Fixed [here](https://github.com/huggingface/accelerate/pull/576/commits/10334600357bdd014acd67bb255884ee10b54fe6) normally.

Reply:
Can confirm this fixes the issue too !